iothub_client/inc/iothub_client_core_common.h

```diff
@@ -234,6 +243,72 @@ extern "C"
         const char* deviceSasToken;
     } IOTHUB_CLIENT_DEVICE_CONFIG;
 
+    /**
+    * @brief    Function callback application implements to process an incoming command from IoT Hub.
+    *
+    * @param[in]    componentName         Name of the component associated with this request.  If this is targeting the root component, this will be NULL.
+    * @param[in]    commandName           Name of the command associated with this request.
+    * @param[in]    payload               Raw payload of the request.  This is NOT guaranteed to be a \0 terminated string.
+    * @param[in]    size                  Number of bytes of @p payload.
+    * @param[in]    payloadContentType    Payload type of @p payload.  This currenly will ALWAYS be "application/json" and applications using the default
+    *                                     options on the IoT Hub client can safely assume they will only receive JSON type.
+    *                                     Future versions of the IoT Hub client may add options that allow alternate contentTypes to be sent via
+    *                                     commands.  This will NOT be the default behavior but will require specific application opt-in.
+    * @param[out]   response              Response generated by the callback that will be sent to IoT Hub.  This must be allocated using malloc().  
+    *                                     The SDK takes ownership of the buffer on callback return and will free() it when the SDK is done with it.
+    * @param[out]   responseSize          Number of bytes application is returning in response.
+    * @param[in]    userContextCallback   User context pointer set in initial call to IoTHubDeviceClient_LL_SubscribeToCommands.
+    *
+    * @remarks   The application behavior is undefined if the user calls
+    *            the IoTHubDeviceClient_LL_Destroy function from within any callback.
+    *
+    * @return    Status code of the method to return to the service.  This maps to an HTTP style status code.
+    */
+    typedef int(*IOTHUB_CLIENT_COMMAND_CALLBACK_ASYNC)(
+                    const char* componentName,
+                    const char* commandName,
+                    const unsigned char* payload,              
+                    size_t size,
+                    const char* payloadContentType,
```

Comment:
>payloadContentType [](http://example.com/codeflow?start=32&length=18)

is this content type of user data ? Or some intermittent encoding done by hub, based on connection to IoTHub?
Like CBor, can customer also send cbor data using command?